### PR TITLE
added numpy.histogram_bin_edges function 

### LINKED ISSUE #19568: histogram_bin_edges

### PR DESCRIPTION
**Pull request by ShreyanshBardia**

### CODE REVIEW

**Comment by zhumakhan** [comment]
Hey, thanks for your PR! Getting shape related errors while running tests. Could you please address these issues?
```
FAILED ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_histograms.py::test_numpy_histogram_bin_edges[cpu-ivy.functional.backends.numpy-False-False] - AssertionError: returned shape = (11,), ground-truth returned shape = (10,)
FAILED ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_histograms.py::test_numpy_histogram_bin_edges[cpu-ivy.functional.backends.jax-False-False] - AssertionError: returned shape = (1226,), ground-truth returned shape = (1224,)
FAILED ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_histograms.py::test_numpy_histogram_bin_edges[cpu-ivy.functional.backends.tensorflow-False-False] - AssertionError: returned shape = (266,), ground-truth returned shape = (265,)
```

**Comment by ShreyanshBardia** [comment]
> Hey, thanks for your PR! Getting shape related errors while running tests. Could you please address these issues?
> 
> ```
> FAILED ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_histograms.py::test_numpy_histogram_bin_edges[cpu-ivy.functional.backends.numpy-False-False] - AssertionError: returned shape = (11,), ground-truth returned shape = (10,)
> FAILED ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_histograms.py::test_numpy_histogram_bin_edges[cpu-ivy.functional.backends.jax-False-False] - AssertionError: returned shape = (1226,), ground-truth returned shape = (1224,)
> FAILED ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_histograms.py::test_numpy_histogram_bin_edges[cpu-ivy.functional.backends.tensorflow-False-False] - AssertionError: returned shape = (266,), ground-truth returned shape = (265,)
> ```

Hi @zhumakha, is this from the latest commit test logs? I checked the latest tests but couldn't find these. To confirm it is supposed to be under `intelligent-tests-pr / run_tests (1) (pull_request) `, right?

**Comment by ShreyanshBardia** [comment]
@zhumakhan can you review this once. The errors occurring, are from the minute difference between ivy.linspace and np.linspace. You can confirm this from the following code block

```
first_edge=-2
last_edge=1.5
np.linspace(first_edge,last_edge,23,endpoint=True)==ivy.linspace(first_edge,last_edge,23,endpoint=True)
```

**Comment by ShreyanshBardia** [comment]
@zhumakhan all the tests are passing now, except for one in paddle which I think is unrelated to the PR

**Comment by ShreyanshBardia** [comment]
Hi @zhumakhan, can you please review this PR

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_histograms.py [hunk 1]
```diff
@@ -43,3 +124,32 @@ def test_numpy_bincount(
...
+        range=range,
+        weights=weights,
+    )
+    ret_gt = np.histogram_bin_edges(
+        a=a,
+        bins=bins,
+        range=range,
+        weights=weights,
+    )
+    assert np.allclose(ret.ivy_array, ret_gt)
```
Comment (zhumakhan) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_histograms.py:144]: Hey, we shouldn't  directly use `allclose` in tests. Instead, please following test writing gudies and examples: https://github.com/unifyai/ivy/blob/main/ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_order_statistics.py#L53

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_histograms.py [hunk 2]
```diff
@@ -43,3 +124,32 @@ def test_numpy_bincount(
...
+
+
+@handle_frontend_test(fn_tree="numpy.histogram", data=_histogram_helper())
+def test_numpy_histogram(data):
```
Comment (zhumakhan) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_histograms.py:148]: Hey, we shouldn't  directly use `allclose` in tests. Instead, please following test writing gudies and examples: https://github.com/unifyai/ivy/blob/main/ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_order_statistics.py#L53

**Review by zhumakhan — CHANGES_REQUESTED** [review]

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_histograms.py (hunk 2 above)
Comment (zhumakhan) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_histograms.py:148]: Here is the docs on writing tests for frontend functions: https://unify.ai/docs/ivy/overview/deep_dive/ivy_frontends_tests.html

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_histograms.py (hunk 1 above)
Comment (ShreyanshBardia) [reply ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_histograms.py]: @zhumakhan I have updated the tests, can you take a look now

**Comment by ShreyanshBardia** [comment]
@zhumakhan it looks like tests are not running for some reason, can you check this please

**Comment by zhumakhan** [comment]
Hey, it seems PR has been updated, could you please respond to comments above with how latest changes address the issues mentioned before?

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_histograms.py [hunk 3]
```diff
@@ -45,3 +123,53 @@ def test_numpy_bincount(
...
+def test_numpy_histogram_bin_edges(
+    data,
+    on_device,
+    fn_tree,
+    frontend,
+    backend_fw,
+    test_flags,
+):
+    input_dtype, a, bins, range, weights, density = data
+    print(input_dtype)
```
Comment (zhumakhan) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_histograms.py:138]: Please remove this print statement

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_histograms.py [hunk 4]
```diff
@@ -45,3 +123,53 @@ def test_numpy_bincount(
         weights=None,
         minlength=0,
     )
+
+
+@handle_frontend_test(fn_tree="numpy.histogram_bin_edges", data=_histogram_helper())
+def test_numpy_histogram_bin_edges(
```
Comment (zhumakhan) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_histograms.py:129]: Part of tests are failing due to bfloat16 is not supported by numpy. Either we should not test for bfloat16 dtype (avoiding it when generating arrays for testing) or handle it explicitly

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_histograms.py [hunk 5]
```diff
@@ -45,3 +123,53 @@ def test_numpy_bincount(
...
+        on_device=on_device,
+        a=a,
+        bins=bins,
+        range=range,
+        weights=weights,
+    )
+
+
+@handle_frontend_test(fn_tree="numpy.histogram", data=_histogram_helper())
+def test_numpy_histogram(
```
Comment (zhumakhan) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_statistics/test_histograms.py:154]: Showing some function related errors: `TypeError: histogram() got multiple values for argument 'density'`

**Review by zhumakhan — CHANGES_REQUESTED** [review]

**Pull request closed without merging**